dockstore-webservice/src/main/java/io/dockstore/webservice/helpers/QuayImageRegistry.java

```diff
@@ -133,11 +158,45 @@ public QuayImageRegistry(final Token quayToken) {
         return tags;
     }
 
-    private Tag convertQuayTagToTag(QuayTag quayTag, Tool tool) throws InvocationTargetException, IllegalAccessException {
+    // For each manifest in the list:
+    // Find the matching Quay tag (using the digest)
+    // Store the collected Image(s) into a map that consists of <Original Manifest List Quay Tag, List<Images>>.
+    // Remove the matching Quay tag from the list so that a Dockstore version is not created based off of it.
+    private List<Image> handleMultiArchQuayTags(Tool tool, QuayTag quayTag, Gson g, List<QuayTag> cleanedQuayTagList) throws ApiException {
+        LOG.info(quayToken.getUsername() + " ======================= Getting image for tag {}================================", quayTag.getName());
+        QuayRepoManifest quayRepoManifest;
+        quayRepoManifest = manifestApi.getRepoManifest(quayTag.getManifestDigest(), tool.getNamespace() + '/' + tool.getName());
+        DockerManifestList manifestList = g.fromJson(quayRepoManifest.getManifestData(), DockerManifestList.class);
+        List<Image> images = new ArrayList<Image>();
+        Arrays.stream(manifestList.getManifests()).forEach(manifest -> {
+            Optional<QuayTag> matchingTag = cleanedQuayTagList.stream().filter(tag -> (tag.getManifestDigest().equals(manifest.getDigest()))).findFirst();
```

Comment:
wonder if its worth programming defensively, so that if one image has an unexpected structure, we move onto the next image/tag

Reply:
still a little curious about this but won't block. If for, example the `manifestDigest` changes format,  `guayTag.getManifestDigest()` is null. or similar won't this die and prevent any tag from being created since the calling method is only handing illegal access and invocation target exceptions?

Reply:
I thought that the try/catch block starting at 176 would cover this? If the string response doesn't follow the format of the `DockerManifestList.class`, then it'd throw the exception and have an empty image list instead.

Reply:
I think that's catching `JsonSyntaxException` so wouldn't catch a valid JSON response but with null properties. For example, if quay.io started returning `DockerManifestList` with null properties, I think the `NullPointerException` would be uncaught.

Maybe I'm a little too paranoid that quay.io might change things up.